pkg/deploy/controller/image_change_controller.go

```diff
@@ -108,16 +135,26 @@ func parseImage(name string) (string, string) {
 	return name[:index], name[index+1:]
 }
 
-func generateTriggerDetails(triggers []deployapi.DeploymentTriggerImageChangeParams) *deployapi.DeploymentDetails {
+func generateTriggerDetails(config *deployapi.DeploymentConfig, triggers []deployapi.DeploymentTriggerImageChangeParams) *deployapi.DeploymentDetails {
 	// Generate the DeploymentCause objects from each DeploymentTriggerImageChangeParams object
 	// Using separate structs to ensure flexibility in the future if these structs need to diverge
 	causes := []*deployapi.DeploymentCause{}
 	for _, trigger := range triggers {
+		// The trigger From field takes precedence over RepositoryName
+		repoName := trigger.RepositoryName
+		if len(trigger.From.Name) > 0 {
+			namespace := trigger.From.Namespace
+			if len(namespace) == 0 {
+				namespace = config.Namespace
+			}
+			repoName = fmt.Sprintf("%s/%s", namespace, trigger.From.Name)
```

Comment:
We can't, when we do v1beta2 we'll nuke it.

Basically, from -> repo.name says "I want to use the Status.DockerImageRepository field on the repo with Repo.Name as my docker pull spec".  I am not saying "pull from a docker repository named Name".  The indirection on ImageRepository is important - a user can then point you to someplace else.

----- Original Message -----
> >  	// Generate the DeploymentCause objects from each
> >  	DeploymentTriggerImageChangeParams object
> >  	// Using separate structs to ensure flexibility in the future if these
> >  	structs need to diverge
> >  	causes := []*deployapi.DeploymentCause{}
> >  	for _, trigger := range triggers {
> > +		// The trigger From field takes precedence over RepositoryName
> > +		repoName := trigger.RepositoryName
> > +		if len(trigger.From.Name) > 0 {
> > +			namespace := trigger.From.Namespace
> > +			if len(namespace) == 0 {
> > +				namespace = config.Namespace
> > +			}
> > +			repoName = fmt.Sprintf("%s/%s", namespace, trigger.From.Name)
> 
> Agree that From should be added to DeploymentCauseImageTrigger, but I'm not
> clear on this:
> 
> >You can't do this - you need to be loading this from
> >ImageRepository.Status.ImageRepository, because the string you get here
> >does not necessarily correlate to what the repository is set up to.
> >From.Name isn't guaranteed to point to a given docker repository.
> 
> If I add From to the cause, I'd be storing Name/Namespace on the cause, and
> no docker repository information at all (it's abstracted away behind the
> ImageRepository). The only slot for Status.ImageRepository would be
> cause.RepositoryName- is it okay to overload that field (e.g. populate both
> From AND RepositoryName resolved from the IR?)
> 
> Having both these fields on the internal API is really messy. Is there any
> way we could do a conversion from Trigger.RepositoryName to Trigger.From so
> that internally we only have Trigger.From?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/858/files#r24096333